### ISSUE TITLE
Deleting notification not updating display AND semi-corrupting data

### ISSUE THREAD

**Issue by msorens** [semgrep/semgrep#2303]
**Describe the bug**

1. Created webhook in https://chefio.slack.com / #auth-team-notify on November 12.
2. Created webhook in https://chefio.slack.com / #a2-notify on December 20.

https://chefio.slack.com/apps/... then correctly showed these two:
<img width="721" alt="image" src="https://user-images.githubusercontent.com/6817500/102731648-3e9b9c80-42ed-11eb-86e6-9b32cab2a5b8.png">

3. In https://semgrep.dev/manage/notifications, I pressed <kbd>Delete</kbd> to remove the earlier notification.
=> First issue: Apparently nothing happened; both still showed.

4. I then refreshed the page.
=> Second issue: the old notification went away, but the second notification appeared twice! (I blanked out for security reasons, but the two URLs are identical.)
<img width="1075" alt="image" src="https://user-images.githubusercontent.com/6817500/102731817-b79af400-42ed-11eb-8f36-571bb6baf01e.png">

5. I then pressed <kbd>Delete</kbd> on one of them and this time the display updated itself, now showing just one.

**To Reproduce**
As above.

**Expected behavior**
Selected notification should have been correctly removed, and the screen updated to reflect this, in step 3.

**Screenshots**
As above.

**What is the priority of the bug to you?**
Is this a P0 (blocking your adoption of Semgrep or workflow), P1 (important to fix or quite annoying), P2 (regular bug that should get fixed)?
P2 regular -- not something I would need again in the near term.

**Environment**
??

**Comment by mschwager** [comment]
Hi @msorens,

Interesting behavior - seems like there could be a race condition somewhere. Unfortunately I'm unable to reproduce.

Here are the exact steps I tried:

- Visit https://semgrep.dev/manage/notifications
- "+ Notification Channel" -> Slack -> paste valid Slack hook URL1 -> Save
- "+ Notification Channel" -> Slack -> paste valid, but different, Slack hook URL2 -> Save
- Delete URL2 -> successfully deleted action
- Delete URL1 -> successfully deleted action

I'm unable to reproduce the failed delete, or the duplicated URL. I also tried with URL1 == URL2, but still no luck. Are you still able to reproduce this behavior?

This sounds like it could be browser/OS dependent. Could you include your system information/versions?

**Comment by msorens** [comment]
Just experimented a bit and I am seeing a new variation, 100% repeatable.
(This is more reliable data since I was paying close attention to my steps carefully this time.) 

My system is a Mac running Mojave (10.14.6).
Same issues in Chrome [Version 87.0.4280.88 (Official Build) (x86_64)] and in Firefox [84.0.1 (64-bit)].

Today I started with my one good, working slack notification and never touched that one.

Steps:

1. I added a new dummy slack notification, e.g., https://hooks.slack.com/services/foo/chrome01
2. I then deleted the notification, it was removed from the page, but I did NOT receive a successful deletion notice.
3. I refreshed the page and that notification was visible again. (Along with my one good one mentioned above.)
4. Refreshing several times over a couple minutes and that supposedly deleted notification persisted.
5. I deleted it again, it was removed from the page, and this time I DID receive a successful deletion notice. And it did NOT reappear upon refreshing.

This seems to be the crucial point: 

To fail ("fail" being defined as (a) no deletion notice and (b) item returns upon refresh): 
Create notification, NO refresh, then delete notification.

To succeed: 
Create notification, do one or more refreshes, then delete notification.
